### ISSUE TITLE
AI lacks the knowledge that European archers can light pitch ditches

### ISSUE THREAD

**Issue by Heroesflorian** [Evrey/SHC_AIV#11]
It is relatively well-known that some AI characters in Crusader do use pitch ditches, and others don't. 

It is less-known that some who have pitch ditches in their vanilla castles (e.g. Abbot) do not use them, while some others without pitch ditches in their vanilla castles (e.g. Saladin) do use them if given any. 

It is known to barely anyone (I remember someone mentioning it below one of Udwin's videos many months ago) that the pitch-usage of AI characters depends only on the availability of: 
- braziers
- Arabian archers

After wondering why the Wolf on M80 didn't really light his pitches when Udwin attacked him, I finally decided to test this out. My long-held (false) belief that the Wolf (and possibly some other characters) just are less "trigger-happy" with their pitch ditches as compared to Saladin or Caliph, and only light when more hostile units are on the pitch ditches, has been overthrown as a result of those tests. 
It seems the pitch ditch usage really is just a binary thing set globally, ignoring the presence of European archers altogether, and not differing between characters at all. 
There are some rare instances of a European archer's fire arrows missing their (moving) targets and then accidentally hitting a pitch ditch instead, same as what can happen for fire throwers or engineers with oil pots as well, but that does not count as "intentional usage". (And btw, speaking of oil pots, there seem to be differences (aggressive vs defensive stance --> activation with 1+ vs 3+ hostile units in range) between different characters, unlike for the pitch ditches.) 

As a consequence of this, it is now clear why e.g. Abbot never uses his vanilla castles' pitch ditches (lacks Arabian archers), Richard or Frederick can never be "forced" to use them (same reason), Snake can occasionally use them if given (Arabian archers on keep / other fortifications when gathering or advancing for a siege), why Wolf is rather defensive with his pitch lighting behavior (can only light from Arabian archers on keep or worker protection patrol groups standing atop a gate temporarily), and why e.g. Nizar can be "taught" pitch ditch usage so easily (has Arabian archers). 

So why is this relevant for your castles, you may ask? Well, e.g. your Wolf castles are designed with the mindset of "European archers can light pitch ditches", which is certainly true for human players, and hence also what you would expect for the AI, but which also certainly is false for the AI in fact. 
A more practical design would place braziers on top of the keep and on those parts of the walls / gates / towers where groups of Arabian archers would move to or past/through most likely, when trying to approach the sites where some of Wolf's workers were killed recently. 

And yep, another bug I also dislike a lot, you are not alone with your likely rage/despair! 

As a side effect, and as weird as it may sound, the "decorative" braziers in this entrance design do actually serve a lot more of a purpose than the braziers on flanking bastions next to a gatehouse, as they allow Arabian archers passing through the entrance area to light pitch ditches from any tile between the center of the inner drawbridge to the inner border outer gatehouse: 

![grafik](https://user-images.githubusercontent.com/12549394/39960053-ca113660-561b-11e8-9b81-67d76b5b96e2.png)

![grafik](https://user-images.githubusercontent.com/12549394/39960075-239193ec-561c-11e8-84af-3cbf5746e41e.png)

Another consequence of this is, it is fairly easy to buff the Sultan, who can not only magically recruit special slaves to dig moats, but also use pitch ditches from his keep (and sometimes walls, when preparing for an attack): 

![grafik](https://user-images.githubusercontent.com/12549394/39960097-a6d0b710-561c-11e8-8797-fc933cf74896.png)

(In the design from which the screenshot below was taken, he also has his keep enclosed VERY fast, way more tightly so than in his vanilla castles, and has some fire and tower ballistae. With that setup, 5 allied Sultans manage to quickly subdue a hostile Wolf (without any Sultans dying) in a 40k starting gold match (or anything less, because Sultan's castle is cheap, Wolf's isn't), whereas 5 vanilla Sultans get completely annihilated and killed by the very same Wolf with an identical setup.)

**Comment by Evrey** [comment]
Great news! =D

... not.

Well, not only the Wolf is affected, also my Abbot.

As for the Abbot, I think I'd just remove all the pitch. And the Wolf, well... Probably the Sheriff way, i.e. only put 'em where oil engineers are nearby, emphasizing the "suicide inferno" of Wolf 2. I think that adding external pitch just for accidental lucky shots of a few patrols is a waste of resources. In addition to that, the Wolf's castles are way too huge to light external pitch from the keep. So... pitch nearby oil engineers and lots of suicide ditches.

**Comment by Heroesflorian** [comment]
You could, however, at least have some internal pitch ditches (as present for wolf4.aiv and wolf7.aiv currently), as those have a fairly high chance of being lit by Arabian archers atop Wolf's keep. Because all AI characters gather their default ranged patrol troops exactly there, and recruit exactly those by preference in a number of situations, including being attacked. But that's up to you of course. 

For the Abbot, well, yeah, either remove all pitch, or just do it "the Sheriff way". Abbot works well (without issues, unlike some other characters) with boiling oil pots.

**Comment by Evrey** [comment]
https://github.com/Sh0wdown/UnofficialCrusaderPatch

This here could - in the long run - make it possible.

**Comment by Heroesflorian** [comment]
Indeed...

**Comment by Evrey** [comment]
I temporarily marked this as **wontfix** until we know what will happen with this:

> - [ ] European bowmen won't set pitch alight
>
> ~ [src](https://github.com/Sh0wdown/UnofficialCrusaderPatch/issues/5)

I.e. blocked on part of that issue.

**Comment by ByBurton** [comment]
This issue should be highlighted.
Wolf could be soo much stronger with this..

**Comment by Evrey** [comment]
https://github.com/Sh0wdown/UnofficialCrusaderPatch/issues/165

Alternatively, this will at least fix the Wolf.

**Comment by Heroesflorian** [comment]
Indeed. Although that'd only be a hack-fix, and leave others (e.g. Abbot) out in the cold, so to say. 

But for Wolf, that would definitely be quite an improvement, not just regarding pitch ditches but also his general defensive strength, making him undoubtedly the deadliest defender to run into...

**Comment by a-good-noob** [comment]
Hey, 
what about giving the Abbot an Oil Smelter ?
In the Vanilla settings, the abbot keeps some pitch in his storage place (No Sell-Resource=Pitch). Therefore, Firefly might thought of giving the Abbot that defence thing. Moreover, in Stronghold Classic you gain the ability to pour pitch while defending the monastery.

All in all, with an Oil Smelter the Abbot could also light up pitch ditches and his weak defence will be strengthened.

**Comment by Heroesflorian** [comment]
Good idea indeed... I went for that with some own Abbot, and it works quite well imho.
